examples/EllipticalSliceSampling.jl

```diff
@@ -0,0 +1,135 @@
+# observed under Gaussian Noise with $\sigma = 0.001$ .
+
+fx = f(x_train, 0.001)
+
+# Data's log-likelihood w.r.t prior `GP`. 
+
+logpdf(fx, y_train)
+
+# Calculating the exact posterior over `f` given `y`. The GP's kernel currently has some
+# arbitrary fixed parameters. 
+
+p_fx = posterior(fx, y_train)
+
+# Data's log-likelihood under the posterior `GP`. We see that it drastically increases.
+
+logpdf(p_fx(x_test), y_test)
+
+# Plot the posterior `p_fx` along with the observations.
+
+plt = scatter(x_train, y_train; label = "Train data")
+scatter!(plt, x_test, y_test; label = "Test data")
+plot!(plt, p_fx, 0:0.001:1; label="Posterior")
+
+# # Elliptical Slice Sampler
+# Previously we computed the log likelihood of the untuned kernel parameters 
+# of the GP, $-1.285$. We now also perform approximate inference over said 
+# kernel parameters using the 
+# [Elliptical Slice Sampling](http://proceedings.mlr.press/v9/murray10a/murray10a.pdf)
+# provided by
+# [EllipticalSliceSampling.jl](https://github.com/TuringLang/EllipticalSliceSampling.jl/).
+# We start of by loading necessary packages.
+
+using EllipticalSliceSampling, Distributions
+
+# We define a function which returns log-probability of the data under the 
+# GP / log-likelihood of the parameters of the GP.
+
+function logp(params; x=x_train, y=y_train)
+    exp_params = exp.(params)
+    kernel = ScaledKernel(
+        transform(
+            Matern52Kernel(), 
+            ScaleTransform(exp_params[1])
+            ), 
+        exp_params[2]
+    )
+    f = GP(kernel)
+    fx = f(x, 0.1)
+    return logpdf(fx, y)
+end
+
+# We define a Gaussian prior over the joint distribution on kernel parameters
+# space. Since we have only two parameters, we define a multi-variate Gaussian
+# of dimension two.
+
+prior = MvNormal(2, 1)
+
+# Sanity check for the defined `logp` function and `prior` distribution.
+
+logp(rand(prior))
+
+# Generate 2,000 samples using `ESS_mcmc` provided by `EllipticalSliceSampling.jl`. 
+
+samples = ESS_mcmc(prior, logp, 2_000);
+samples_mat = hcat(samples...)';
+
+# Mean of samples of both the parameters.
+
+mean_params = mean(samples_mat, dims=1)
+
+# Plot a histogram of the samples for the two parameters. The vertical line in each 
+# graph indicates the mean of the samples. 
+
+plt = histogram(samples_mat; layout=2, labels="Param")
+vline!(plt, mean_params; layout=2, label="Mean")
+
+# Average log-marginal-probability of data with posterior kernel parameter samples 
+# sampled using ESS. We can observe that there is significant improvement over 
+# exact posterior with default kernel parameters.
+
+mean([logp(param; x=x_test, y=y_test) for param in samples])
+
+# Plot sampled functions from posterior with tuned parameters
+
+
+plt = scatter(x_train, y_train; label="Train data")
+scatter!(plt, x_train, y_train; label="Test data")
+for params in eachrow(samples_mat[end-100:end,:])
```

Comment:
Maybe
```suggestion
for params in @view(samples[(end-100):end])
```
or
```suggestion
for i in 1_900:2_000
    params = samples[i]
```
is a bit simpler?